tests/test_extractors.py

```diff
@@ -91,3 +91,15 @@ def test_extract_from_directory(self, extractor, tmp_path, file):
     def test_extract_from_nonsense(self, extractor):
         with pytest.raises(NotImplementedError):
             extractor.extract(8)
+
+    @pytest.mark.parametrize(
+        "source",
+        [
+            """
+                def foo():
+            """,
+        ],
+    )
+    def test_extract_with_syntax_error(self, extractor, file_path, file, caplog):
```

Comment:
**suggestion (testing):** Consider adding more syntax error scenarios to the test.

The current test case covers a very basic syntax error scenario. It would be beneficial to include more complex syntax errors, such as missing colons, incorrect indentation, or invalid syntax within a function body, to ensure the extractor handles a wide range of syntax errors gracefully.